src/python/turicreate/toolkits/object_detector/object_detector.py

```diff
@@ -1644,7 +1644,7 @@ def export_coreml(self, filename,
         return self.__proxy__.export_to_coreml(filename, options)
 
 
-    def predict(self, dataset):
+    def predict(self, dataset, confidence_threshold=0.25, iou_threshold=0.45):
```

Comment:
Where did we get these defaults from?

Reply:
I follow the default values from Python's implementation.

Reply:
But if you are asking where did the values come from in the beginning,
I think they are just empirical results.

Reply:
It seems like for the `evaluate()`, the default value for `confidence_threshold` and `iou_threshold` are both `None`, and for the `predict()`, the default values are 0.25 and `None`, respectively. hmmmm

Reply:
I think `0.45` is the way to go. Explicit is definitely the pythonic way.

Reply:
Okay, let's go with that for now, since it's a better way to express the actual functionality supported by 5.8. But worth considering: the current Python implementation of the toolkit unofficially supports setting the `iou_threshold` once during model creation, and then defaulting to that threshold for all future calls to `predict` and `evaluate`. With this approach, a user who wants a different threshold will never be able to encode that into the model itself.

Reply:
A relevant question: is the IOU threshold for non-maximum suppression, which is _only_ relevant for inference, something that belongs in the trained model itself?

Reply:
@nickjong This is a good point,
somehow in the `init_training()`,
we set the default value for noms to 0.45 :P .
```
  add_or_update_state({
      { "annotations", annotations_column_name },
      { "feature", image_column_name },
      { "model", "darknet-yolo" },
      { "non_maximum_suppression_threshold",
        DEFAULT_NON_MAXIMUM_SUPPRESSION_THRESHOLD },
  })
```
And we never use it if we change the `predict` and `evaluate` function to have input for the you_threshold :P .
Do you think we should delete this state?

Reply:
Note that in `export_to_coreml()`,
it also enable use to set both thresholds :P hmmm.
So I feel like they don't belong to model itself :P .

Reply:
Let's file a followup issue to delete this state as superfluous. Thinking more about it, I think we want to encourage a workflow where somebody trains a model and then uses some kind of model-understanding process to choose the thresholds. So they shouldn't be established by training itself.

Reply:
https://github.com/apple/turicreate/issues/2506